v2/tools/generator/internal/astmodel/type_definition.go

```diff
@@ -135,7 +136,9 @@ func (def TypeDefinition) HasTestCases() bool {
 // FileNameHint returns what a file that contains this name (if any) should be called
 // this is not always used as we often combine multiple definitions into one file
 func FileNameHint(name TypeName) string {
-	return transformToSnakeCase(name.name)
+	result := transformToSnakeCase(name.name)
+	result = strings.Replace(result, "_statusarm", "_status_arm", -1)
```

Comment:
Are there constants we could build these out of, rather than hardcoding them?

Reply:
We do now. And I've used them.